### ISSUE TITLE
Add Source and Ref By Fields

### ISSUE THREAD

**Issue by smiths** [JacquesCarette/Drasil#424]
As mentioned in #418, we should have source and ref by fields.  Rather than URLs we should use citations in the source field.

**Comment by szymczdm** [comment]
Duplicate of #331. Keeping this one as the other does not mention what to include in the source field.

**Comment by szymczdm** [comment]
Source field is now able to be populated. RefBy will require the updated referencing (#361, #35, #47, etc.) to be completed.

**Comment by halonazhao** [comment]
Only the theoretical model had a url link as source and I've already replaced that with a citation. But GM2 doesn't have a source field. IM1 only has a statement as its source. Do I have to do something with them?

**Comment by smiths** [comment]
@halonazhao, this issue is not about noPCM; it is more general.  If this comment is about #425, you should say it there.  If you have changed all existing urls to BibTeX entries, then we can probably close that other issue.

**Comment by halonazhao** [comment]
@szymczdm I think Source and Ref By field have been added into each section. You may want to close this issue?

**Comment by samm82** [comment]
Bumping @halonazhao's question

**Comment by szymczdm** [comment]
Source and RefBy fields are indeed added, however, I don't believe they have been properly populated. The RefBy fields will likely need to wait on traceability. However, I think that can be a separate issue and this can be closed.

Does that sound reasonable @JacquesCarette ?

**Comment by JacquesCarette** [comment]
Yes, quite reasonable. Maybe @samm82 can figure out from the manual version what Source information needs to be added (and create an issue for it), and @halonazhao can create an issue for automatically populating the RefBy fields. Once that's done, this issue can be closed.

**Comment by samm82** [comment]
I created the issue (#997), so this issue can be closed with @halonazhao's issue. @JacquesCarette since this is an easy fix, I just assigned myself and will make the change in all the examples.

**Comment by JacquesCarette** [comment]
@halonazhao is there an issue for automatically populating the RefBy fields?

**Comment by halonazhao** [comment]
The design for filling the `RefBy` is in place. But the problem is still that we need the design of the references to be complete. Currently, collecting all the references is blocking this process. @JacquesCarette

**Comment by JacquesCarette** [comment]
I am thus going to close this issue. The `RefBy` field does exist in the right places, which is what this issue asks for. There are enough other issues about referencing, we're not going to forget to implement it!
